source/adapters/level_zero/device.cpp

```diff
@@ -658,7 +669,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(
         }
       }
     }
-    return ReturnValue(FreeMemory);
+    return ReturnValue(std::min(GlobalMemSize, FreeMemory));
```

Comment:
why is that we want to return the min?

Reply:
We want to return the size up to global memsize, not larger than.

The Free memory could exceed the Global Memory size since Global Memory Size is a calculation, but to avoid OOM condition, we want to have the minimum of the two ie GlobalMemSize > Free Memory, then return Free Memory as the Free Memory Available, but if GlobalMemSize < Free Memory, then return the Global Memory Size as the max "free memory" available.